### ISSUE TITLE
Add an "Inverse Sort" property to BoxContainer

### ISSUE THREAD

**Issue by donut-on-ice** [godotengine/godot-proposals#804]
**Describe the project you are working on:**
Simple GUI with symmetric horizontal lists of overlapping components.

**Describe the problem or limitation you are having in your project:**
When using the property "Separation" of a Box Container with negative values the right/bottom item overlaps the left/top one. 

**Describe the feature / enhancement and how it helps to overcome the problem or limitation:**
There should be a checkbox to inverse sort items in Box Containers so that the items higher in the list will be placed in the right/bottom and not just left/top.

**Describe how your proposal will work, with code, pseudocode, mockups, and/or diagrams:**
Lets say there are 3 items in a VBoxContainer in this order: "(1) (2) (3)"
Lets say the Separation is set to -2/3 of an item's length.
Inverse Sort [  ] -> Box Container will function just like it does now and will display "(((3)"
Inverse Sort [x] -> Box Container will place direct children from the end of the rectangle to the beginning and will display "(3)))"

**If this enhancement will not be used often, can it be worked around with a few lines of script?:**
Probably with 2 lines of code and flipping the resource images but since there is no "z" for control items it seems more natural to have this option in a Box Container

**Is there a reason why this should be core and not an add-on in the asset library?:**
Just that an user would expect this option to be in a Box Container when he would realize there is no "z" property for control items and he would need to keep the order of the items on screen

**Comment by bojidar-bg** [comment]
Was also discussed in https://github.com/godotengine/godot/pull/35791and https://github.com/godotengine/godot/issues/7692.

**Comment by donut-on-ice** [comment]
Indeed this issue is fully covered by https://github.com/godotengine/godot/pull/35791. I will close this

**Comment by bojidar-bg** [comment]
Reopening since the linked PR is not merged yet; also we prefer to slowly move discussions here instead of keeping them on the main repo :smiley:

**Comment by Kazox61** [comment]
I have the same problem as I described in #10980

**Comment by geekley** [comment]
This is a very simple feature that users [expect](https://www.reddit.com/r/godot/comments/1c7xdww/vboxcontainers_how_to_show_children_in_reverse/) from this control. I was quite confused trying to change the alignment property (because it's the only specific field of BoxContainer) because I just assumed it would naturally have to have this feature.

This is why I really think godot's "top-left-centric" approach to 2D, where top-left is treated specially and the other alignments are an afterthought, is fundamentally bad design. It would be much better if it was a "center-centric" approach as I discussed on [this proposal](https://github.com/godotengine/godot-proposals/discussions/9719); where center is special/default and all corners are of equal (un)importance.

You see this issue in several places, e.g. TextureRect's `stretch_mode`, which has just the options center and top-left (arbitrarily chosen as "default") treated as a special corner. What about the other corners? Why should top-left be special in any way from a UI design standpoint? It's not; this is an arbitrary quirk coming from an implementation detail; and it too often gets in the way of designers just wanting to do things the simple intuitive way.

**Comment by Sasino97** [comment]
> This is a very simple feature that users [expect](https://www.reddit.com/r/godot/comments/1c7xdww/vboxcontainers_how_to_show_children_in_reverse/) from this control. I was quite confused trying to change the alignment property (because it's the only specific field of BoxContainer) because I just assumed it would naturally have to have this feature.

Yes, I was also surprised today that it doesn't support it! I want a VBoxContainer to draw the items from the bottom to the top because it's bottom anchored, I would like to avoid changing the children order

**Comment by Kazox61** [comment]
I think this is needed since you can't change the order of the InputEvents. Often you have overlapping elements where the higher control node should consume the events first. With a rerverse sort, you can have control nodes lower in the scene tree but higher in the position. For example the bookmark buttons:

<img width="538" height="475" alt="Image" src="https://github.com/user-attachments/assets/fc78867b-ed4b-410e-a7a0-22ed67386eb3" />

Also the logic for the inverse sorting is already there for RTL support. So why don't we make it also configurable for other use cases?
